tests/Microsoft.DotNet.Docker.Tests/EnvironmentVariableInfo.cs

```diff
@@ -0,0 +1,102 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Xunit;
+
+namespace Microsoft.DotNet.Docker.Tests
+{
+    public class EnvironmentVariableInfo
+    {
+        public string ActualValue { get; private set; }
+        public bool AllowAnyValue { get; private set; }
+        public string ExpectedValue { get; private set; }
+        public string Name { get; private set; }
+
+        public EnvironmentVariableInfo(string name, string expectedValue)
+        {
+            this.Name = name;
+            this.ExpectedValue = expectedValue;
+        }
+
+        public EnvironmentVariableInfo(string name, bool allowAnyValue)
+        {
+            this.Name = name;
+            this.AllowAnyValue = allowAnyValue;
+        }
+
+        public static void Verify(
+            IEnumerable<EnvironmentVariableInfo> variables,
+            DotNetImageType imageType,
+            ImageData imageData,
+            DockerHelper dockerHelper)
+        {
+            GetActualValues(variables, imageType, imageData, dockerHelper);
+
+            foreach (EnvironmentVariableInfo variable in variables)
+            {
+                if (variable.AllowAnyValue)
+                {
+                    Assert.NotEmpty(variable.ActualValue);
+                }
+                else
+                {
+                    Assert.Equal(variable.ExpectedValue, variable.ActualValue);
+                }
+            }
+        }
+
+        private static void GetActualValues(
+            IEnumerable<EnvironmentVariableInfo> variables,
+            DotNetImageType imageType,
+            ImageData imageData,
+            DockerHelper dockerHelper)
+        {
+            const char delimeter = '|';
+            
+            IEnumerable<string> echoParts;
+            string invokeCommand;
+            char delimiterEscape;
+            if (DockerHelper.IsLinuxContainerModeEnabled)
+            {
+                echoParts = variables.Select(envVar => $"${envVar.Name}");
+                invokeCommand = $"/bin/sh -c";
+                delimiterEscape='\\';
+            }
+            else
+            {
+                echoParts = variables.Select(envVar => $"%{envVar.Name}%");
+                invokeCommand = $"CMD /S /C";
+                delimiterEscape='^';
+            }
+
+            string echoArgs = echoParts.Aggregate((working, next) => $"{working}{delimiterEscape}{delimeter}{next}");
```

Comment:
```suggestion
            string echoArgs = String.Join(delimiterEscape + delimiter, echoParts);
```

This achieves the same result less verbosely.